### ISSUE TITLE
Filter for regular expression test returns matches but procmail returns negative match

### ISSUE THREAD

**Issue by abclution** [webmin/usermin#60]
Total procmail newbie here..having some issues.

This is a negative match filter, essentially used to filter and forward any emails send to this address, that are NOT from "@mygreatdomain.com"

I can only get it to match when I use, Return-Path header, but even matching in Usermin, it fails during procmail execution.

`
^[<A-Za-z0-9._%+-]+@(?!mygreatdomain.com)[A-Za-z0-9.-]+\.[A-Za-z]{2,4}[>]$`

```
procmail: [20447] Thu Jul 16 04:54:38 2020
procmail: No match on "^Return-Path: .*^[<A-Za-z0-9._%+-]+@(?!mygreatdomain.com)[A-Za-z0-9.-]+\.[A-Za-z]{2,4}[>]$.*"
procmail: Assigning "LASTFOLDER=/home/mygreatdomain..com/homes/test/Maildir/new/1594864478.20447_0.vps1"
procmail: Notified comsat: "test-mygreatdomain..com@0:/home/mygreatdomain..com/homes/test/Maildir/new/1594864478.20447_0.vps1"
From user@gmail.com  Thu Jul 16 04:54:34 2020
 Subject: asdfasdfasdfasd
  Folder: /home/mygreatdomain..com/homes/test/Maildir/new/1594864478.20447_0.     4634
procmail: Executing "/etc/webmin/virtual-server/procmail-logger.pl"
Time:1594864478 From:andreas.tyrosvoutis@gmail.com To:test@mygreatdomain..com User:test-mygreatdomain..com Size:4695 Dest:/home/mygreatdomain..com/homes/test/Maildir/new/1594864478.20447_0.vps1 Mode:None

```

**Comment by jcameron** [comment]
Why not create a filter first that matches all messages from mygreatdomain.com and delivers to the inbox, then a second filter that forwards everything? That way you don't need to worry about negative matches.

**Comment by abclution** [comment]
Hi @jcameron 
Sorry for the late response.

In the end, I took your suggestion. I don't believe that procmails version of extended regular expression works with negative lookahead anyways. It can use variables and comparators, but it turns out in the end your suggestion was the easiest way to accomplish what I needed. Significantly less complicated 👍 Thank you for that. I was stuck trying to do it the specific way I had envisioned it. I tried to move the mountain instead of walk around it..

I spent literally days trying to grok procmail and its quirks but documentation regarding procmail regular expression is uh, to put it lightly difficult to understand and no central "authoritative" definition of its quirks. + Regular expression newbie = ouch

The only thing I can mention is that the validation of the test filters is a bit off, giving me matches that procmail says no to. This is likely exactly the same issue..."procmails version of extended regular expression + quirks" the only true validator for it, is itself. ie procmail 💣 

Cool cool cool. 

Closing and thanks for the help.
